### ISSUE TITLE
Lost settings option

### ISSUE THREAD

**Issue by richd67** [hkgnp/logseq-todoist-plugin#7]
I incorrectly installed the plugin using my Todoist API and now in installed plugins I get a star rather than the settings icon and thus it is impossible to uninstall and start again?

**Comment by hkgnp** [comment]
Hi!
Could you take a screen shot please?

**Comment by richd67** [comment]
![2021-11-04 (2)](https://user-images.githubusercontent.com/70855766/140329757-ee6da73e-2e27-4f43-8466-9434a839b938.png)

**Comment by hkgnp** [comment]
Oh! You are actually on the wrong tab; you are at the Marketplace. 

Please click on the "Installed Plugins" tab, next to the "Marketplace" tab and you should be all set.

**Comment by hkgnp** [comment]
Hmmm I am not sure why this is the case. Do you mind restating Logseq to see if the issue gets resolved?

**Comment by richd67** [comment]
I have restarted it several times with no effect. I also cannot see any files associated with the plugin in the logseq folder, I was assuming that there would be something there I could delete?

**Comment by hkgnp** [comment]
> Actually that is not the case this is my installed tab, the marketplace has many more plugins than listed here.

Apologies I am on my phone and I misread the screenshot. 

Could you see the post above and see if it helps please? If not, I would think that going to the plug-ins directory in your Logseq folder to delete manually would be the other thing I would try.

**Comment by hkgnp** [comment]
> I have restarted it several times with no effect. I also cannot see any files associated with the plugin in the logseq folder, I was assuming that there would be something there I could delete?

Are you on a Mac? The plugins are usually kept in ~/.logseq, which is a hidden file. I am not clear at the moment where it is stored on a Windows machine but I can try to find out.

**Comment by richd67** [comment]
Yes found it, its in my windows profile page? Do you suggest deleting the whole directory?

**Comment by hkgnp** [comment]
> Yes found it, its in my windows profile page? Do you suggest deleting the whole directory?

I would suggest deleting the specific folder in your plugins directory. The folder would be called `logseq-todoist-plugin`.

**Comment by richd67** [comment]
Yes that worked, will try and reload, this time reading the instructions a bit more carefully.

**Comment by hkgnp** [comment]
> Yes that worked, will try and reload, this time reading the instructions a bit more carefully.
> 
> 

Phew! Glad to hear that. So sorry about the experience. I'm not sure how it could have happened too since the settings is just an external file outside of the graph. 

Do let me know how else I can help!

**Comment by richd67** [comment]
Reloaded everything and followed the instructions but back in the same place, not setting icon in the installed plugin tab and no push or pull from Todoist.

**Comment by hkgnp** [comment]
Oh dear! 

Perhaps I should ask, what version of Logseq are you using? I'll try to get a Windows machine to do some testing on (I assume you are using Windows?).

**Comment by richd67** [comment]
I am running version 0.4.5 and yes running windows. I hadnt realised the some of the plugins were stored on my windows/user/  path as all of my other logseq files are in my Gdrive folder and some of the plugins also live there. I would have thought that they would all be found in the same folder? Not sure if this is part of the issue?

**Comment by hkgnp** [comment]
> I am running version 0.4.5 and yes running windows. I hadnt realised the some of the plugins were stored on my windows/user/  path as all of my other logseq files are in my Gdrive folder and some of the plugins also live there. I would have thought that they would all be found in the same folder? Not sure if this is part of the issue?

I am not sure! I just tested with a Windows machine and didn't encounter the same error. I would think all the plug-ins installed from the Marketplace would be stored in the same place whereas plug-ins installed separately may not. 

I've posted this in the Discord channel to see if any of the other developers have any clues as to why this happened. 

So sorry about the experience! If you really need to use it, you can try the manual approach. If anyone from the Discord channel replies, I'll be sure to look into it!

**Comment by richd67** [comment]
Thanks, I can see you post of Discord, lets hope someone has some suggestion for the error

**Comment by richd67** [comment]
Tried a manual install and ran into some errors with the npm build?
G:\My Drive\logseq\logseq-todoist-plugin\logseq-todoist-plugin>npm install && npm run build
npm notice
npm notice New patch version of npm available! 8.1.0 -> 8.1.3
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.1.3
npm notice Run npm install -g npm@8.1.3 to update!
npm notice
npm ERR! code ENOTDIR
npm ERR! syscall mkdir
npm ERR! path G:\My Drive\logseq\logseq-todoist-plugin\logseq-todoist-plugin\node_modules\path\node_modules
npm ERR! errno -4052
npm ERR! ENOTDIR: not a directory, mkdir 'G:\My Drive\logseq\logseq-todoist-plugin\logseq-todoist-plugin\node_modules\path\node_modules'

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\richa\AppData\Local\npm-cache\_logs\2021-11-05T16_23_18_091Z-debug.log


I then updated to latest ver and got same errors
Not sure what to do next as this is not a logseq issue

**Comment by hkgnp** [comment]
Hi @richd67!

The directory paths look a little odd. Perhaps you can use the below instructions instead:

1. Download the latest release from https://github.com/hkgnp/logseq-todoist-plugin/releases/tag/v0.91
2. Unzip the file into anywhere of your choice. For example, as indicated above: `G:\My Drive\logseq`. This will create a folder called `logseq-todoist-plugin`.
3. Create the .env file as per the Manual Instructions on the [Readme](https://github.com/hkgnp/logseq-todoist-plugin/blob/master/README.md).
4. Only after creating the .env file in the same folder as your `package.json` file, you go to your terminal, and run `npm install`. See if you have any errors. If you have no errors, then you run `npm run build`.
5. If the above is successful, you can then go to Logseq to manually install the plugin.

**Comment by hkgnp** [comment]
Hey @richd67!

Someone on Discord faced the same issue, but am not sure if it is the same plugin. But anyway, he had a fix for his situation that you may want to try.

Essentially, you go to your settings folder, and delete any duplicate looking settings. Suggest to remove any `logseq-todoist-plugin` that you currently have installed first, before deleting. That way you can try re-installing it again, and see if it helps. Please note that the latest release in the marketplace is quite different from previous versions, and use the slash command to pull tasks, instead of clicking on the button.

Source: https://discord.com/channels/725182569297215569/735747000649252894/906803205000810507

**Comment by richd67** [comment]
Similar outcome
G:\My Drive\logseq\logseq-todoist-plugin-v0.91-alpha\logseq-todoist-plugin-v0.91-alpha>npm install
npm ERR! code ENOTDIR
npm ERR! syscall mkdir
npm ERR! path G:\My Drive\logseq\logseq-todoist-plugin-v0.91-alpha\logseq-todoist-plugin-v0.91-alpha\node_modules\css-modules-loader-core\node_modules
npm ERR! errno -4052
npm ERR! ENOTDIR: not a directory, mkdir 'G:\My Drive\logseq\logseq-todoist-plugin-v0.91-alpha\logseq-todoist-plugin-v0.91-alpha\node_modules\css-modules-loader-core\node_modules'

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\richa\AppData\Local\npm-cache\_logs\2021-11-07T19_27_15_283Z-debug.log

G:\My Drive\logseq\logseq-todoist-plugin-v0.91-alpha\logseq-todoist-plugin-v0.91-alph
[2021-11-07T19_27_15_283Z-debug.log](https://github.com/hkgnp/logseq-todoist-plugin/files/7492952/2021-11-07T19_27_15_283Z-debug.log)
a>

**Comment by hkgnp** [comment]
Hey! 

Thanks for providing the error log. 

If there is a 'tmp' folder in the plug-ins or Logseq-todoist directory, please delete. 

Then run 'npm cache clean --force' and try again?

**Comment by richd67** [comment]
Re installed V2 push seems to be working fine but for pull I getting the message function not available on home page that it only works on journal page but I am on a journal page with date at top?? What am I missing here?

**Comment by richd67** [comment]
OK i have worked it out now, so does not work on a journal page only normal pages, that might mean a rethink of my normal workflow?

**Comment by hkgnp** [comment]
> Re installed V2 push seems to be working fine but for pull I getting the message function not available on home page that it only works on journal page but I am on a journal page with date at top?? What am I missing here?

Hi! Glad to hear that it works for you now!

Based on the feedback from another user, it only becomes a journal page when you click on the date itself.

However, I am curious because v2 does not make use of the button anymore, but instead uses the `/pull` command. So I am not sure if you are on v2 yet. In v2, the button actually opens up a popup for you to key in your variables. Please see the latest instructions here: https://github.com/hkgnp/logseq-todoist-plugin/blob/master/README.md

**Comment by richd67** [comment]
Yes on V2 and am using/pull. The addition of the button to add variables is great. I have subsequently gone in and altered the settings file manually but the button is a great start point.
Yes clicking on the date works.

**Comment by richd67** [comment]
Just noticed when you pull it disappears from Todoist and is marked done? Could there be an option to keep it there until it is completed in Logseq?

**Comment by richd67** [comment]
YEs hard to please everyone and have solutions for every workflow, but great work anyhow.

**Comment by hkgnp** [comment]
> YEs hard to please everyone and have solutions for every workflow, but great work anyhow.

I've added the fix so you can just update the plugin. After which, you can pull tasks from the home page. 

Thanks for your kind words! Will be closing the issue but please let me know if you encounter any more problems!
